stripe/src/main/java/com/stripe/android/paymentsheet/CurrencyFormatter.kt

```diff
@@ -9,19 +9,35 @@ import kotlin.math.pow
 internal class CurrencyFormatter {
     fun format(
         amount: Long,
-        currency: Currency
+        currencyAmount: Currency,
+        targetLocale: Locale = Locale.getDefault()
     ): String {
-        val currencySymbol = currency.getSymbol(Locale.getDefault())
+        val currencySymbol = currencyAmount.getSymbol(targetLocale)
         val majorUnitAmount =
-            amount / MAJOR_UNIT_BASE.pow(getDefaultFractionDigits(currency).toDouble())
-        val currencyFormat = NumberFormat.getCurrencyInstance()
+            amount / MAJOR_UNIT_BASE.pow(getDefaultFractionDigits(currencyAmount).toDouble())
+
+        // The language string found in the system properties prevents an extra space being
+        // added after the currency symbol.  If it can't be found we will default to the display locale
+        val targetLanguageFormat = System.getProperty("user.language.format", targetLocale.language)
```

Comment:
I'm not sure if I exactly understand the difference between this vs what we get from `Locale.getDefault`. How often would we expect these properties to be populated?

Maybe we can do this for testability:

```
internal class CurrencyFormatter(
    private val propertyValue: (String, String) -> String? = { key, defaultValue ->
        System.getProperty(key, defaultValue)
    }
) {
    fun format(
        amount: Long,
        currencyAmount: Currency,
        targetLocale: Locale = Locale.getDefault(),
    ): String {
       // ...
       val targetLanguageFormat = propertyValue("user.language.format", targetLocale.language)
       // ...
    }
}
```

Can we add `.takeUnless { it.isNullOrBlank() }` to all of these?